ldcbench.api/src/main/java/org/dice_research/ldcbench/rdf/SimpleTripleCreator.java

```diff
@@ -56,9 +71,23 @@ public Triple createTriple(int sourceId, int propertyId, int targetId, int targe
      *            it is an internal node
      * @param isProperty
      *            a flag indicating whether the node is a property
+     * @param isBlankNode
+     *            a flag indicating whether the node is a blankNode
+     * @param isLiteral
+     *            a flag indicating whether the node is a literal
      * @return the created {@link Node} instance
      */
-    public Node createNode(int nodeId, int externalId, int extGraphId, boolean isProperty) {
+    public Node createNode(int nodeId, int externalId, int extGraphId, boolean isProperty,
+            boolean isBlankNode, boolean isLiteral) {
+        Node n;
+        if (isBlankNode) {
+            n = NodeFactory.createBlankNode(String.valueOf(nodeId));
+            return n;
+        }
+        if (isLiteral) {
+            n = NodeFactory.createBlankNode(String.format(UriHelper.LITERAL, nodeId));
```

Comment:
I see a small issue here. The literal attribute is defined as `LITERAL = "literal"`. The `nodeId` won't play any role in the generated literal string. Hence, all literals will end up with the same string representation. That should be changed.

Reply:
This problem hasn't been fixed.